torch/csrc/jit/passes/peephole.cpp

```diff
@@ -188,16 +188,24 @@ void PeepholeOptimizeImpl(Block* block, bool addmm_fusion_enabled) {
       // replace the output with true if node is __isnot__ or false if node is
       // __is__
       AT_ASSERT(node->inputs().size() == 2);
-      for (size_t check_none_index : {0, 1}) {
-        bool input_must_be_none =
-            node->inputs().at(check_none_index)->mustBeNone();
-        bool other_must_not_be_none =
-            node->inputs().at(1 - check_none_index)->mustNotBeNone();
-        if (input_must_be_none && other_must_not_be_none) {
-          WithInsertPoint guard(node);
-          auto output = node->owningGraph()->insertConstant(
-              node->kind() == aten::__isnot__);
-          node->output()->replaceAllUsesWith(output);
+      if (node->inputs().at(0)->mustBeNone() &&
```

Comment:
This works and we probably don't want to further complicate the PR... I think the more general case is in change constant propagation so that it knows `prim::AutogradZero` is a constant, but i'm not sure how much that would buy us in optimization anyways. you can ignore this comment lol

Reply:
Well, I'm not sure an argument is necessarily a constant (though arguably None is), so constant propagation feels a bit wrong here. But I'm very happy to move it.

Reply:
I think it's good in